src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -324,6 +324,11 @@ export const useSearch = (services: DiscoverViewServices) => {
       const savedSearchInstance = await getSavedSearchById(savedSearchId);
       setSavedSearch(savedSearchInstance);
 
+      // if saved search does not exist, do not atempt to sync filters and query from savedObject
+      if (!savedSearch) {
```

Comment:
i think the linter will complain that this isn't included in the dependency right? but then 325 sets this value.

i can see in the future someone listening to the linter and including the savedsearch in the dependencies.

shouldn't we just do 
```typescript
if (!savedSearchInstance) {
```

Reply:
changed to use `savedSearchId` instead since it is already a dependency

Reply:
savedSearchInstance does not work here because it is always defined

Reply:
`savedSearchId` here wont work as well, because when we create a new saved search, the id is null, but we do not want to return here.